### PR TITLE
feat(iam): add applications tab

### PR DESCRIPTION
**Pull request by rjamet-ovh**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-15410
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [ ] Commits are signed-off
- [ ] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Improvement from external contributor

### CODE REVIEW

File: packages/manager/modules/iam/src/components/deleteEntity/deleteEntity.controller.js [hunk 1]
```diff
@@ -87,6 +87,8 @@ export default class DeleteEntityController {
       promise = this.deletePolicy();
     } else if (this.entity.type === ENTITY.RESOURCE_GROUP) {
       promise = this.deleteResourceGroup();
+    } else if (this.entity.type === ENTITY.APPLICATION) {
+      promise = this.deleteApplication();
     } else if (this.entity.type === ENTITY.RESOURCE_TYPE) {
       promise = this.$q.when(true);
     } else {
```
Comment (tibs245) [inline packages/manager/modules/iam/src/components/deleteEntity/deleteEntity.controller.js:94]: Maybe begin to replace by catch to readability and anticipate evolution ?

Comment (tibs245) [reply packages/manager/modules/iam/src/components/deleteEntity/deleteEntity.controller.js:94]: switch/case sorry

File: packages/manager/modules/iam/src/components/deleteEntity/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -12,6 +12,12 @@
   "iam_delete_entity_resourceGroup_name": "Suppression du groupe de ressources : ",
   "iam_delete_entity_resourceGroup_success": "Le groupe de ressources a bien été supprimé",
   "iam_delete_entity_resourceGroup_warn": "Êtes-vous sûr.e de confirmer la suppression ?",
+  "iam_delete_entity_application_error": "An error occurred while deleting the application: {{ message }}",
+  "iam_delete_entity_application_field": "Please enter the word \"{{ statement }}\" to delete your application",
+  "iam_delete_entity_application_heading": "Delete an application",
+  "iam_delete_entity_application_name": "Deleting application: ",
+  "iam_delete_entity_application_success": "Application deleted",
+  "iam_delete_entity_application_warn": "Warning: this action will delete the application and it will revoke all credential belonging to this application.",
```
Comment (tibs245) [inline packages/manager/modules/iam/src/components/deleteEntity/translations/Messages_fr_FR.json:20]: It's normal the text is in english ?

**Review by darsene — APPROVED** [review]

File: packages/manager/modules/iam/src/components/deleteEntity/translations/Messages_fr_FR.json (hunk 2 above)
Comment (rjamet-ovh) [reply packages/manager/modules/iam/src/components/deleteEntity/translations/Messages_fr_FR.json:20]: yes, i'll fix the translations content on a regular PR which will be based on this PR target branch

File: packages/manager/modules/iam/src/dashboard/applications/applications.module.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import routing from './applications.routing';
+import deleteModule from './delete';
+
+const moduleName = 'ovhManagerIAMDashboardApplications';
+
+angular
+  .module(moduleName, [deleteModule])
+  .component('iamApplications', component)
+  .config(routing)
+  .run(/* @ngTranslationsInject:json ./translations */);
```
Comment (tibs245) [inline packages/manager/modules/iam/src/dashboard/applications/applications.module.js:13]: No translation folder, so this line is useless (?)

File: packages/manager/modules/iam/src/components/deleteEntity/translations/Messages_fr_FR.json (hunk 2 above)
Comment (darsene) [reply packages/manager/modules/iam/src/components/deleteEntity/translations/Messages_fr_FR.json:20]: Will be fixed in a next PR when it's merged

File: packages/manager/modules/iam/src/components/deleteEntity/deleteEntity.controller.js (hunk 1 above)
Comment (darsene) [reply packages/manager/modules/iam/src/components/deleteEntity/deleteEntity.controller.js:94]: Could be fixed in a next PR when it's merged

File: packages/manager/modules/iam/src/dashboard/applications/applications.module.js (hunk 3 above)
Comment (darsene) [reply packages/manager/modules/iam/src/dashboard/applications/applications.module.js:13]: Will be fixed in a next PR when it's merged

File: packages/manager/modules/iam/src/dashboard/applications/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  "iam_applications_datagrid_column_application_id": "Application id",
+  "iam_applications_datagrid_column_application_name": "Application name",
+  "iam_applications_datagrid_column_application_description": "Description",
+  "iam_applications_datagrid_column_application_status": "Status",
+  "iam_applications_datagrid_column_application_status_active": "Active",
+  "iam_applications_datagrid_column_application_status_inactive": "Inactive",
+  "iam_applications_datagrid_column_application_key": "Application key",
+  "iam_applications_datagrid_action_delete_application": "Delete application",
+  "iam_applications_description": "View and manage your applications. For more information about applications, see the documentation."
+}
```
Comment (tibs245) [inline packages/manager/modules/iam/src/dashboard/applications/translations/Messages_fr_FR.json:11]: English in fr translatation file ?

Comment (darsene) [reply packages/manager/modules/iam/src/dashboard/applications/translations/Messages_fr_FR.json:11]: Will be translated with the next pull request

**Review by tibs245 — APPROVED** [review]

**Pull request merged**